cluster/manifests/kube-dns/node-local-daemonset.yaml

```diff
@@ -0,0 +1,99 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: dnsmasq-node
+  namespace: kube-system
+  labels:
+    application: dnsmasq-node
+spec:
+  selector:
+    matchLabels:
+      application: dnsmasq-node
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        application: dnsmasq-node
+    spec:
+      priorityClassName: system-node-critical
+      serviceAccountName: system
+      tolerations:
+      - operator: Exists
+        effect: NoSchedule
+      - operator: Exists
+        effect: NoExecute
+      volumes:
+      - name: kube-dns-config
+        configMap:
+          name: kube-dns
+          optional: true
+      containers:
+      - name: dnsmasq
+        image: registry.opensource.zalan.do/teapot/k8s-dns-dnsmasq-nanny-amd64:1.14.10
+        securityContext:
+          privileged: true
+        livenessProbe:
+          httpGet:
+            path: /healthcheck/dnsmasq
+            port: 9054
+            scheme: HTTP
+          initialDelaySeconds: 60
+          timeoutSeconds: 5
+          successThreshold: 1
+          failureThreshold: 5
+        args:
+        - -v=2
+        - -logtostderr
+        - -configDir=/etc/k8s/dns/dnsmasq-nanny
+        - -restartDnsmasq=true
+        - --
+        - -k
+        - --cache-size=10000
+        - --no-negcache
+        - --server=/cluster.local/10.3.0.10#53
+        - --server=/in-addr.arpa/10.3.0.10#53
+        - --server=/ip6.arpa/10.3.0.10#53
+        ports:
+        - containerPort: 53
+          name: dns
+          protocol: UDP
+        - containerPort: 53
+          name: dns-tcp
+          protocol: TCP
+        # see: https://github.com/kubernetes/kubernetes/issues/29055 for details
+        resources:
+          requests:
+            cpu: 20m
+            memory: 25Mi
+        volumeMounts:
+        - name: kube-dns-config
+          mountPath: /etc/k8s/dns/dnsmasq-nanny
+      - name: sidecar
+        image: registry.opensource.zalan.do/teapot/k8s-dns-sidecar-amd64:1.14.10
+        securityContext:
+          privileged: true
+        livenessProbe:
+          httpGet:
+            path: /metrics
```

Comment:
@szuecs what about changing this to not do a liveness probe on the metrics endpoint which is quite a lot of data? (my comment in the other PR probably got lost).

Reply:
I replied before that this is the default and I am not sure if it makes sense to change it. 
Maybe they thought about it, maybe not....

Reply:
Probably not :)

But we can optimize later.